### PR TITLE
update: added logs 

### LINKED ISSUE #137: feat: add more debug and trace logs in code
Currently it's pretty difficult to debug broken code because there are very few logs, we need to add more

### PR DESCRIPTION
**Pull request by heemankv**
# This PR resolves #137.

### CODE REVIEW

**Review by heemankv — COMMENTED** [review]
Done with Self Review

File: crates/orchestrator/src/tests/common/mod.rs [hunk 1]
```diff
@@ -73,10 +73,10 @@ pub async fn create_sqs_queues(provider_config: Arc<ProviderConfig>) -> color_ey
     // Dropping sqs queues
     let list_queues_output = sqs_client.list_queues().send().await?;
     let queue_urls = list_queues_output.queue_urls();
-    log::debug!("Found {} queues", queue_urls.len());
+    tracing::debug!("Found {} queues", queue_urls.len());
     for queue_url in queue_urls {
         match sqs_client.delete_queue().queue_url(queue_url).send().await {
-            Ok(_) => log::debug!("Successfully deleted queue: {}", queue_url),
+            Ok(_) => tracing::debug!("Successfully deleted queue: {}", queue_url),
             Err(e) => eprintln!("Error deleting queue {}: {:?}", queue_url, e),
```
Comment (Mohiiit) [inline crates/orchestrator/src/tests/common/mod.rs:80]: shall we have a warn or error tracing here?

File: crates/prover-services/sharp-service/src/client.rs [hunk 2]
```diff
@@ -72,19 +73,23 @@ impl SharpClient {
         // Adding params to the URL
         add_params_to_url(&mut base_url, params);
 
+        tracing::debug!("Sharp Service: Sending PIE to SHARP");
```
Comment (Mohiiit) [inline crates/prover-services/sharp-service/src/client.rs:76]:
```suggestion
        tracing::info!("Sharp Service: Sending PIE to SHARP");
```

File: crates/prover-services/sharp-service/src/client.rs [hunk 3]
```diff
@@ -99,6 +104,7 @@ impl SharpClient {
         add_params_to_url(&mut base_url, params);
 
         let res = self.client.post(base_url).send().await.map_err(SharpError::GetJobStatusFailure)?;
+        tracing::info!("Sharp Service: Job status response: {:?}", res);
```
Comment (Mohiiit) [inline crates/prover-services/sharp-service/src/client.rs:107]: maybe add trace below where we have error or status okay? wdyt?

File: crates/prover-services/sharp-service/src/lib.rs [hunk 4]
```diff
@@ -43,24 +45,38 @@ impl ProverClient for SharpProverService {
...
-            CairoJobStatus::FAILED => Ok(TaskStatus::Failed(res.error_log.unwrap_or_default())),
+            CairoJobStatus::FAILED => {
+                let error_log = res.error_log.unwrap_or_default();
+                tracing::error!("Sharp Service: Task failed: {:?}", error_log);
+                Ok(TaskStatus::Failed(error_log))
+            }
             CairoJobStatus::INVALID => {
-                Ok(TaskStatus::Failed(format!("Task is invalid: {:?}", res.invalid_reason.unwrap_or_default())))
+                let invalid_reason = res.invalid_reason.unwrap_or_default();
+                tracing::error!("Sharp Service: Task is invalid: {:?}", invalid_reason.clone());
```
Comment (Mohiiit) [inline crates/prover-services/sharp-service/src/lib.rs:61]: do we need clone here?

File: crates/settlement-clients/ethereum/src/lib.rs [hunk 5]
```diff
@@ -154,11 +164,13 @@ impl SettlementClient for EthereumSettlementClient {
         onchain_data_hash: [u8; 32],
         onchain_data_size: [u8; 32],
     ) -> Result<String> {
+        tracing::info!("Ethereum Settlement Client: Updating state with calldata");
         let program_output: Vec<U256> = vec_u8_32_to_vec_u256(program_output.as_slice())?;
         let onchain_data_hash: U256 = slice_u8_to_u256(&onchain_data_hash)?;
         let onchain_data_size = U256::from_be_bytes(onchain_data_size);
         let tx_receipt =
             self.core_contract_client.update_state(program_output, onchain_data_hash, onchain_data_size).await?;
+        tracing::info!("Ethereum Settlement Client: Updated state with calldata {:?}", tx_receipt.transaction_hash);
```
Comment (Mohiiit) [inline crates/settlement-clients/ethereum/src/lib.rs:173]: message not relevant, need to update this

File: crates/settlement-clients/starknet/src/lib.rs [hunk 6]
```diff
@@ -116,32 +117,43 @@ impl SettlementClient for StarknetSettlementClient {
...
+                    tx_hash,
+                    execution_result.revert_reason().unwrap()
+                )))
+            }
             TransactionExecutionStatus::Succeeded => {
                 if tx_receipt.block.is_pending() {
+                    tracing::info!("Transaction {} is pending", tx_hash);
                     Ok(SettlementVerificationStatus::Pending)
                 } else {
+                    tracing::info!("Transaction {} is settled", tx_hash);
```
Comment (Mohiiit) [inline crates/settlement-clients/starknet/src/lib.rs:156]:
```suggestion
                    tracing::info!("Transaction {} is verified", tx_hash);
```

File: crates/settlement-clients/starknet/src/lib.rs [hunk 7]
```diff
@@ -164,17 +176,20 @@ impl SettlementClient for StarknetSettlementClient {
...
         sleep(duration_to_wait_between_polling).await;
+        tracing::info!("Waiting for tx finality for tx hash {:?}", tx_hash);
 
         let tx_hash = Felt::from_hex(tx_hash)?;
         loop {
             let tx_receipt = self.account.provider().get_transaction_receipt(tx_hash).await?;
             if tx_receipt.block.is_pending() {
                 retries += 1;
                 if retries > MAX_RETRIES_VERIFY_TX_FINALITY {
+                    tracing::error!("Max retries exceeeded while waiting for tx {tx_hash} finality.");
```
Comment (Mohiiit) [inline crates/settlement-clients/starknet/src/lib.rs:187]:
```suggestion
                    tracing::error!("Max retries exceeeded while waiting for tx {:?} finality.", tx_hash);
```

File: crates/settlement-clients/starknet/src/lib.rs [hunk 8]
```diff
@@ -164,17 +176,20 @@ impl SettlementClient for StarknetSettlementClient {
...
                     return Err(eyre!("Max retries exceeeded while waiting for tx {tx_hash} finality."));
                 }
                 sleep(duration_to_wait_between_polling).await;
             } else {
+                tracing::info!("Tx {tx_hash} has achieved finality");
```
Comment (Mohiiit) [inline crates/settlement-clients/starknet/src/lib.rs:192]:
```suggestion
                tracing::info!("Tx {:?} has achieved finality", tx_hash);
```

**Review by Mohiiit — CHANGES_REQUESTED** [review]
I have less context on code so maybe you would be able to make a better decision on whether to use info or debug on some places.

**Comment by apoorvsadana** [comment]
Closing in favor of #147

**Pull request closed without merging**